src/Microsoft.AspNetCore.Razor.Evolution/Intermediate/SetPreallocatedTagHelperPropertyIRNode.cs

```diff
@@ -21,7 +21,9 @@ public class SetPreallocatedTagHelperPropertyIRNode : RazorIRNode
 
         public string PropertyName { get; set; }
 
-        public TagHelperAttributeDescriptor Descriptor { get; set; }
+        public BoundAttributeDescriptor Descriptor { get; set; }
```

Comment:
Spoke offline. Going to add  a reference to `TagHelperBinding` aka `TagBindingResult`